config/config.go

```diff
@@ -306,3 +306,90 @@ const (
 	catchupValidationModeVerifyTransactionSignatures = 4
 	catchupValidationModeVerifyApplyData             = 8
 )
+
+// SaveConfigurableConsensus saves the configurable protocols file to the provided data directory.
+// if the params contains zero protocols, the existing consensus.json file will be removed if exists.
+func SaveConfigurableConsensus(dataDirectory string, params ConsensusProtocols) error {
```

Comment:
I thought this was moved in another PR, but fine otherwise.

Reply:
I moved everything that had to move because it couldn't compile in the SDK, then I moved one extra function because it was clearly related to the previously moved stuff, it just didn't happen to _require_ moving because it didn't use any other types.  I did that in different commits, but I think it's all in the PR.